std/container/array.d

```diff
@@ -539,11 +525,13 @@ Complexity: $(BIGOH 1)
     }
 
 /**
-Ensures sufficient capacity to accommodate $(D e) elements.
+Ensures sufficient capacity to accommodate $(D e) _elements.
+If $(D e < capacity), this method will not decrease the capacity.
 
 Postcondition: $(D capacity >= e)
 
-Complexity: $(BIGOH 1)
+Complexity: $(BIGOH e) if $(D e > capacity),
```

Comment:
This code should have amortized O(1) complexity if you ask me. The reasoning is as follows: reserve just *makes sure* there is enough capacity but the internal algorithm of Array, should, of course, increase capacity using some proportional progression so as to minimize reallocations.

In other words reserve doesn't set the capacity to value lower then the actual capacity of the array already is so the code should behave as amortized complexity prescribes. It's a bug if it doesn't.